src/services/plugins/socket.rs

```diff
@@ -72,27 +67,28 @@ impl SocketService {
 
     async fn handle_connection<
```

Comment:
Not sure have we stress tested this part. Probably we would need improvements by using connection pools

Reply:
not stress tested yet, I'm planning to do it this week